src/iocore/aio/AIO.cc

```diff
@@ -378,15 +400,27 @@ cache_op(AIOCallbackInternal *op)
 #ifdef AIO_FAULT_INJECTION
           err = aioFaultInjection.pread(a->aio_fildes, (static_cast<char *>(a->aio_buf)) + res, a->aio_nbytes - res,
                                         a->aio_offset + res);
+#else
+#if TS_USE_MMAP
+          ink_assert((static_cast<char const *>(a->aio_fildes.first) + a->aio_offset + a->aio_nbytes) <= a->aio_fildes.last);
```

Comment:
ink_assert will only check in debug builds.  I think we should return an error in `op->aio_result` in case of an out-of-bounds access.

Reply:
Would replacing it with something like this be enough in your opinion?

#if TS_USE_MMAP
          if ((static_cast<char const *>(a->aio_fildes.first) + a->aio_offset + a->aio_nbytes) <= a->aio_fildes.last) {
            memcpy(static_cast<char *>(a->aio_buf) + res, static_cast<char const *>(a->aio_fildes.first) + a->aio_offset + res,
                 err = a->aio_nbytes - res);
          } else {
            err = -1;
          }

This will be similarly as for other AIO modes (pwrite and pread).

Reply:
Fixed - f41723afadcb36e6d1a502fb1575e092e8afc9c4